cg/services/illumina/backup/utils.py

```diff
@@ -0,0 +1,66 @@
+"""Helper functions."""
+
+from datetime import datetime
+from operator import attrgetter
+
+from cg.constants import FileExtensions
+from cg.services.illumina.backup.models import PdcEncryptionKey, PdcSequencingFile
+
+
+class DsmcOutput:
+    DATE_COLUMN_INDEX = 2
+    TIME_COLUMN_INDEX = 3
+    PATH_COLUMN_INDEX = 4
+
+
+def is_pdc_encryption_key(line: str) -> bool:
+    return (
+        FileExtensions.KEY in line
+        and FileExtensions.GPG in line
+        and FileExtensions.GZIP not in line
+    )
+
+
+def is_pdc_sequencing_path(line: str) -> bool:
+    return FileExtensions.TAR in line and FileExtensions.GZIP in line and FileExtensions.GPG in line
+
+
+def get_latest_pdc_archived_sequencing_run(
+    pdc_files: list[PdcSequencingFile],
+) -> PdcSequencingFile | None:
+    """Return the most recent sequencing run based on the 'dateTime' attribute, or None if no files are present."""
+
+    if not pdc_files:
+        return None
+
+    latest_file = max(pdc_files, key=attrgetter("dateTime"))
+
+    return latest_file
+
+
+def get_latest_pdc_encryption_key(pdc_files: list[PdcEncryptionKey]) -> PdcEncryptionKey | None:
+    """Return the most recent encryption key based on the 'dateTime' attribute, or None if no keys are present."""
+
+    if not pdc_files:
+        return None
+
+    latest_file = max(pdc_files, key=attrgetter("dateTime"))
+
+    return latest_file
+
+
+def convert_string_to_datetime_object(strDateTime: str) -> datetime:
+    date_formats = [
+        "%Y-%m-%d %H:%M:%S",
+        "%Y/%m/%d %H:%M:%S",
+        "%d/%m/%Y %H:%M:%S",
+        "%m/%d/%Y %H:%M:%S",
+    ]
+
+    for fmt in date_formats:
+        try:
+            return datetime.strptime(strDateTime, fmt)
+        except ValueError:
+            continue
+
+    raise ValueError(f"Could not convert '{strDateTime}' to a datetime object.")
```

Comment:
check cg/utils/time.py

Reply:
I like this function 😅. Would it be okay to move it to cg/utils/time.py, or would you suggest using the existing functions instead? If so, I’ll need to rethink my approach a bit.

Reply:
I did a new commit. I hope its better, I did not find the functionality in time.py or date.py, so I added a refined version of my script in date.py.